File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int count=0;
        for(int i=0;i<nums.length-1;i++){
            int sum=nums[i];
            if(sum==k) count++;
            for(int j=i+1;j<nums.length;j++){
                sum+=nums[j];
                if(sum==k) count++;
            }
        }
        if(nums[nums.length-1]==k) count++;
        return count;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& arr, int k) {
        int n = arr.size();
        map<long long, int> mp;

        int ans = 0;
        long long sum = 0;

        for (int i = 0; i < n; i++) {
            sum += arr[i];

            if (sum == k)
                ans++;

            long long target = sum - k;

            if (mp.count(target)) {
                ans+= mp[target];
            }

            
            mp[sum]++;
        }

        return ans;
    }
};